mongo/dataset_store.go

```diff
@@ -139,40 +120,19 @@ func (m *Mongo) GetEditions(ctx context.Context, id, state string, offset, limit
 	defer s.Close()
 
 	selector := buildEditionsQuery(id, state, authorised)
-	q := s.DB(m.Database).C(editionsCollection).Find(selector)
+	q := s.DB(m.Database).C(editionsCollection).Find(selector).Sort()
 
-	totalCount, err := q.Count()
+	// get total count and paginated values according to provided offset and limit
+	results := []*models.EditionUpdate{}
+	totalCount, err := QueryPage(ctx, q, offset, limit, &results)
 	if err != nil {
```

Comment:
Note: the old code error handling was a bit contradictory (ignore ErrNotFound, but if totalCount < 1 then return an error) - I assumed that we want an error to be returned if we have 0 items in this case.